src/mysql.erl

```diff
@@ -437,6 +438,49 @@ execute_transaction(Conn, Fun, Args, Retries) ->
             {aborted, Aborted}
     end.
 
+-spec change_user(Conn, Username, Password) -> Result
+    when Conn :: connection(),
+         Username :: iodata(),
+         Password :: iodata(),
+         Result :: ok | {error, server_reason}.
+change_user(Conn, Username, Password) ->
+    gen_server:call(Conn, {change_user, Username, Password, undefined}).
+
+-spec change_user(Conn, Username, Password, Options) -> Result
+    when Conn :: connection(),
+         Username :: iodata(),
+         Password :: iodata(),
+         Options :: [Option],
+         Result :: ok | {error, server_reason},
+         Option :: {database, iodata()}
+                 | {queries, [iodata()]}
+                 | {prepare, [NamedStatement]},
+         NamedStatement :: {StatementName :: atom(), Statement :: iodata()}.
+change_user(Conn, Username, Password, Options) ->
+    Database = proplists:get_value(database, Options, undefined),
+    Ret = gen_server:call(Conn, {change_user, Username, Password, Database}),
+    case Ret of
+        ok ->
+            Queries = proplists:get_value(queries, Options, []),
+            lists:foreach(fun (Query) ->
+                              case query(Conn, Query) of
+                                  ok -> ok;
+                                  {ok, _} -> ok;
```

Comment:
Some duplicated code from handshake code here? Please move shared code out to a function.

Reply:
Yes, I just wanted to make a rough sketch of it first to show you before I make it pretty ;) That will be straightened out in the final version, of course.

Reply:
Code duplication has been removed with my latest commit, moved into `execute_after_connect/3` (not a very imaginative name, I admit.